viz_scripts/plots.py

```diff
@@ -78,16 +85,39 @@ def pie_chart_mode(plot_title,labels,values,file_name):
 
     m_labels, m_values = merge_small_entries(labels, values)
     
-    def func(pct, values):
-        total = sum(values)
-        absolute = int(round(pct*total/100.0))
-        return "{:.1f}%\n({:d})".format(pct, absolute) if pct > 4 else''
- 
     wedges, texts, autotexts = ax.pie(m_values,
                                       labels = m_labels,
                                       colors=[colours[key] for key in labels],
                                       pctdistance=0.75,
-                                      autopct= lambda pct: func(pct, values),
+                                      autopct= lambda pct: format_pct(pct, values),
+                                      textprops={'size': 23})
+
+    ax.set_title(plot_title, size=25)
+    plt.text(-1.3,-1.3,f"Last updated {arrow.get()}", fontsize=10)
+    plt.setp(autotexts, **{'color':'white', 'weight':'bold', 'fontsize':20})
+    plt.savefig(SAVE_DIR+file_name+".png", bbox_inches='tight')
+    plt.show()
+
+def pie_chart_sensed_mode(plot_title,labels,values,file_name):
```

Comment:
Would it be a good idea to refactor pie_chart_sensed_mode() and pie_chart_mode() functions - extract common code and make call from these functions?
Except for all_labels, I see all these code to be identical.

Reply:
Yes. But I would want to create an automated test harness first to ensure that there are no regressions.
And we would likely do some refactoring/cleanup as part of the basemode changes anyway - `plots.py` in particular has multiple copies of the same function, just with different labels, all of which should be collapsed to one as part of this change or an immediate cleanup.